<launch>
  <!-- Left -->
  <node name="apriltag_detector_left" pkg="apriltag_ros" type="apriltag_ros_continuous_node" output="screen">
    <remap from="/image_rect" to="/spot/camera/left/image"/>
    <remap from="/camera_info" to="/spot/camera/left/camera_info"/>

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_settings.yaml"/>
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_tags.yaml"/>
  </node>

  <!-- Right -->
  <node name="apriltag_detector_right" pkg="apriltag_ros" type="apriltag_ros_continuous_node" output="screen">
    <remap from="/image_rect" to="/spot/camera/right/image"/>
    <remap from="/camera_info" to="/spot/camera/right/camera_info"/>

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_settings.yaml"/>
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_tags.yaml"/>
  </node>

  <!-- Back -->
  <node name="apriltag_detector_back" pkg="apriltag_ros" type="apriltag_ros_continuous_node" output="screen">
    <remap from="/image_rect" to="/spot/camera/back/image"/>
    <remap from="/camera_info" to="/spot/camera/back/camera_info"/>

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_settings.yaml"/>
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_tags.yaml"/>
  </node>

  <!-- Frontleft -->
  <node name="apriltag_detector_frontleft" pkg="apriltag_ros" type="apriltag_ros_continuous_node" output="screen">
    <remap from="/image_rect" to="/spot/camera/frontleft/image"/>
    <remap from="/camera_info" to="/spot/camera/frontleft/camera_info"/>

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_settings.yaml"/>
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_tags.yaml"/>
  </node>

  <!-- Frontright -->
  <node name="apriltag_detector_frontright" pkg="apriltag_ros" type="apriltag_ros_continuous_node" output="screen">
    <remap from="/image_rect" to="/spot/camera/frontright/image"/>
    <remap from="/camera_info" to="/spot/camera/frontright/camera_info"/>

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_settings.yaml"/>
    <rosparam command="load" file="$(find spot_navigation)/config/apriltag_tags.yaml"/>
  </node>



</launch>
